qiskit/algorithms/evolvers/variational/variational_principles/variational_principle.py

```diff
@@ -0,0 +1,83 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for a Variational Principle."""
+
+from abc import ABC, abstractmethod
+from typing import Union, List, Callable
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import (
+    CircuitQFI,
+    CircuitGradient,
+    StateFn,
+    ListOp,
+    OperatorBase,
+)
+
+
+class VariationalPrinciple(ABC):
+    """Class for a Variational Principle."""
```

Comment:
Can any of the docs here be expanded upon a bit to help out those that might not be quite so familiar with things. Its rather minimal here and states nothing more really than the Python tells you i.e.  `class VariationalPrincipal`